tasks/grunt-release.js

```diff
@@ -301,9 +301,11 @@ module.exports = function(grunt){
           if (tasks.length) {
             grunt.log.ok('running ' + taskName + ' ');
             if (!nowrite) {
+              var promisedTasks = new Array(tasks.length);
               for (var i = 0; i < tasks.length; i++) {
-                run('grunt ' + tasks[i], '-> ' + tasks[i]);
+                promisedTasks[i] = run('grunt ' + tasks[i], '-> ' + tasks[i]);
```

Comment:
I'm just a spectator (I was about to create my own merge request when I've saw yours), but I don't think it's necessary to store the tasks in an associative array. A simpler `var promisedTasks = [];` and `promisedTasks.push(run( ... ));` would do the trick.

Another improvement: the entire `runTasks` method can be simplified to something like this:
```javascript
    function runTasks(taskName) {
      var tasks = options[taskName];
      var promises = [];

      if (tasks.length) {
        grunt.log.ok('running ' + taskName + ' ');
        if (!nowrite) {
          for (var i = 0; i < tasks.length; i++) {
            promises.push(run('grunt ' + tasks[i], '-> ' + tasks[i]));
          }
        }

      return Q.all(promises);
    }
```
No need for the `Q.fcall` wrapper because the `Q.all` will return a promise too.